### ISSUE TITLE
Concatenation operation

### ISSUE THREAD

**Issue by maxvanspengler** [maxvanspengler/hyperbolic_learning_library#41]
A frequently used operation from the PyTorch library is [torch.cat](https://pytorch.org/docs/stable/generated/torch.cat.html). This operation can trivially be applied to ManifoldTensor objects when dim != man_dim, but this is a bit more complicated when concatenation happens in the manifold dimension. For example, in the Poincaré ball we should use beta-concatenation as described in the [HNN++ paper](https://arxiv.org/pdf/2006.08210.pdf). 

This concatenation operation would preferably be defined on every manifold object, so for now on the Euclidean and PoincareBall manifold classes. An example of beta-concatenation can be found in the unfold method of the PoincareBall class.

**Comment by diegogcerdas** [comment]
Hello, I am working on it in PR #50 😃

**Comment by diegogcerdas** [comment]
Are we good to close this issue?

**Comment by maxvanspengler** [comment]
Oops, missed this. Yes we are!
